packages/klighd-core/src/views-styles.tsx

```diff
@@ -394,13 +392,64 @@ export function colorDefinition(colorId: string, start: ColorStyle, end: ColorSt
             ...(end.opacity ? {'stop-opacity': end.opacity} : {})
         }}
     />
-    const angleFloat = angle === undefined ? 0 : angle
+    let angleFloat = angle === undefined ? 0 : angle
+
+    // Calculate the x and y lengths a line of angle 'angle' would have in a 1x1 box.
+    
+    // First, normalize the angle to be 0<=angle<360
+    angleFloat = angleFloat % 360
+    if (angleFloat < 0) {
+        angleFloat += 360
+    }
+    // Convert the angle to radians
+    const angleRad = angleFloat / 180 * Math.PI
+
+    let x, y
+    if (angleRad <= 1/4 * Math.PI || angleRad > 7/4 * Math.PI) {
+        x = 1
+        y = - Math.tan(0/2 * Math.PI - angleRad)
+    } else if (angleRad <= 3/4 * Math.PI) {
+        x = Math.tan(1/2 * Math.PI - angleRad)
+        y = 1
+    } else if (angleRad <= 5/4 * Math.PI) {
+        x = -1
+        y = Math.tan(2/2 * Math.PI - angleRad)
+    } else { // or: else if (angleRad <= 7/4 * Math.PI) {
+        x = - Math.tan(3/2 * Math.PI - angleRad)
+        y = -1
+    }
+
```

Comment:
shouldn't this all directly be solvable with `x = Math.cos(angleRad)` and `y = Math.sin(angleRad)`?

Reply:
Ok I understand, looks good to me then.